src/System.Devices.Gpio/System.Devices.Gpio/UnixSpiDevice.cs

```diff
@@ -138,6 +138,18 @@ private unsafe void Initialize()
             }
         }
 
+        public override void Read(byte[] buffer)
+        {
+            byte[] writeBuffer = new byte[buffer.Length];
```

Comment:
Allocations are not free.  We shouldn't be allocated unnecessarily or doing a R+W when only a "half duplex" operation was requested.